kit/src/main/java/com/oracle/javafx/scenebuilder/kit/editor/EditorPlatform.java

```diff
@@ -331,54 +344,44 @@ public static void open(String path) throws IOException {
         }
 
         if (!args.isEmpty()) {
-            executeDaemon(args, null);
+            executeDaemon(args, null, 0);
         }
     }
-
+    
     /**
-     * Requests the underlying platform to "reveal" the specified folder. On
-     * Linux, it runs 'nautilus'. On Mac, it runs 'open'. On Windows, it runs
-     * 'explorer /select'.
+     * Requests the underlying platform to open the specified folder in its default file system viewer. This will reveal any file therein.
+     * On Linux, it runs {@code 'xdg-open'}. On Mac, it runs {@code 'open'} and on Windows it runs {@code 'explorer /select'}.
      *
      * @param filePath path for the folder to be revealed
-     * @throws IOException if an error occurs
+     * @throws FileBrowserRevealException This exception allows to catch exits codes != 0 from the called process.
+     * @throws IOException General IOExceptions are thrown by Java System Call Processes in case of an error. 
      */
-    public static void revealInFileBrowser(File filePath) throws IOException {
+    public static void revealInFileBrowser(File filePath) throws IOException, FileBrowserRevealException {
         List<String> args = new ArrayList<>();
-        String path = filePath.toURI().toURL().toExternalForm();
+        String path = Paths.get(filePath.toURI()).normalize().toAbsolutePath().toString();
+        int exitCodeOk = 0;
         if (EditorPlatform.IS_MAC) {
             args.add("open"); //NOI18N
             args.add("-R"); //NOI18N
             args.add(path);
         } else if (EditorPlatform.IS_WINDOWS) {
             args.add("explorer"); //NOI18N
             args.add("/select," + path); //NOI18N
-        } else if (EditorPlatform.IS_LINUX) {
-            // nautilus does fine on Ubuntu, which is a Debian.
-            // I've no idea how it does with other Linux flavors.
-            args.add("nautilus"); //NOI18N
-            // The nautilus that comes with Ubuntu up to 11.04 included doesn't
-            // take a file path as parameter (you get an error popup), you must
-            // provide a dir path.
-            // Starting with Ubuntu 11.10 (the first based on kernel 3.x) a
-            // file path is well managed.
-            int osVersionNumerical = Integer.parseInt(System.getProperty("os.version").substring(0, 1)); //NOI18N
-            if (osVersionNumerical < 3) {
-                // Case Ubuntu 10.04 to 11.04: What you provide to nautilus is
-                // the name of the directory containing the file you want to see
-                // listed. See DTL-5384.
-                path = filePath.getAbsoluteFile().getParent();
-                if (path == null) {
-                    path = "."; //NOI18N
-                }
+            exitCodeOk = 1;
+        } else if (EditorPlatform.IS_LINUX) {           
+            args.add("xdg-open"); //NOI18N
```

Comment:
I was consideriing testing if xdg is installed. However, that might work in different ways for different distributions. Having Nautilus as a backup might work but I think its not really desireable. There are many other file managers out there and `xdg-open` nicely delegates this task to the right point.

However, `xdg-open` is used to open all the documentation URLs, why should we not rely on this to open the file system view? Having a fallback included makes the code more complex without solving the actual problem. `xdg-open` was invented for this.

If we look for fallbacks, than we should not consider Nautilus or any other executable file manager, instead we should look into [alternatives to xdg-open](https://wiki.archlinux.org/title/Default_applications#Resource_openers).

I see multiple options.
* we could enhance the RPM package so, that it has a dependency to `xdg-utils`
* we could make the file manager call configurable in preferences, so that users can update it accordingly to their needs.
* add alternative resource openers on Linux to Scene Builder

My personal favourite would be an option to provide a custom command line call for this task (regardless if Mac, Windows or Linux). Windows is also bound to explorer despite nice alternative file managers such es FreeCommander or TotalCommander.

Reply:
I'm fine removing nautilus, and now I see that `xdg-utils` is already installed in the linux workflow:
https://github.com/gluonhq/scenebuilder/blob/master/.github/workflows/bundles-linux.yml#L30

so I'm okay not having a fallback, but I'd like to show an error dialog if `xdg-utils` is not found nonetheless, instead of the current hidden exception when the command line fails.

As a follow-up, yes, we could add a custom command line call in preferences